src/engine/strat_engine/metadata.rs

```diff
@@ -184,13 +184,14 @@ impl StaticHeader {
 
 
     /// Read data from f into buffer of length _BDA_STATIC_HDR_SIZE.
-    fn read_into_buf<F>(f: &mut F) -> EngineResult<[u8; _BDA_STATIC_HDR_SIZE]>
+    /// Return a newly constructed buffer and the number of bytes read.
+    fn read_into_buf<F>(f: &mut F) -> EngineResult<([u8; _BDA_STATIC_HDR_SIZE], usize)>
         where F: Read + Seek
     {
         try!(f.seek(SeekFrom::Start(0)));
         let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
-        try!(f.read(&mut buf));
-        Ok(buf)
+        let bytes = try!(f.read(&mut buf));
```

Comment:
Match on read_into_buf returning Err in determine_ownership and return DevOwnership::Theirs instead?

If you do stick with this PR's current approach, it really is a change from "read all expected bytes into a buffer or fail" semantics to "read as much as I can and return that", yes? If so, it's likely that functions shouldn't be taking or returning fixed size arrays any longer, but Vec<u8>s and slices, which track their own length. Shouldn't need `bytes`, instead just call `.len()` on the vec or slice.